lib/errors/base-error.ts

```diff
@@ -8,10 +21,10 @@
  * The Base Error all Sequelize Errors inherit from.
  */
 class BaseError extends Error {
-  constructor(message) {
+  constructor(message: string) {
```

Comment:
Given that we were previously calling `super()`, I wonder if this should be `message?: string`?